addons/account/wizard/account_advance_payment.py

```diff
@@ -0,0 +1,339 @@
+        if self.advance_payment_method == 'fixed':
+            delta_amount = (invoice.amount_total - self.fixed_amount) * (-1 if invoice.is_inbound() else 1) * self.order_ids._sign()
+            if not order.currency_id.is_zero(delta_amount):
+                receivable_line = invoice.line_ids.filtered(lambda aml: aml.account_id.account_type == self._get_payment_term_account_type())[:1]
+                product_lines = invoice.line_ids.filtered(lambda aml: aml.display_type == 'product')
+                tax_lines = invoice.line_ids.filtered(lambda aml: aml.tax_line_id.amount_type not in (False, 'fixed'))
+
+                if product_lines and tax_lines and receivable_line:
+                    line_commands = [Command.update(receivable_line.id, {
+                        'amount_currency': receivable_line.amount_currency + delta_amount,
+                    })]
+                    delta_sign = 1 if delta_amount > 0 else -1
+                    for lines, attr, sign in (
+                        (product_lines, 'price_total', -1),
+                        (tax_lines, 'amount_currency', 1),
+                    ):
+                        remaining = delta_amount
+                        lines_len = len(lines)
+                        for line in lines:
+                            if order.currency_id.compare_amounts(remaining, 0) != delta_sign:
+                                break
+                            amt = delta_sign * max(
+                                order.currency_id.rounding,
+                                abs(order.currency_id.round(remaining / lines_len)),
+                            )
+                            remaining -= amt
+                            line_commands.append(Command.update(line.id, {attr: line[attr] + amt * sign}))
+                    invoice.line_ids = line_commands
+
+        # Unsudo the invoice after creation if not already sudoed
+        invoice = invoice.sudo(self.env.su)
+
+        return invoice
+
+    def _prepare_down_payment_section_values(self, order):
+        return {
+            'product_uom_qty': 0.0,
+            'order_id': order.id,
+            'display_type': 'line_section',
+            'is_downpayment': True,
+            'sequence': order.order_line and order.order_line[-1].sequence + 1 or 10,
+        }
+
+    def _prepare_down_payment_lines_values(self, order):
+        """ Create one down payment line per tax or unique taxes combination and per account.
+            Apply the tax(es) to their respective lines.
+
+            :param order: Order for which the down payment lines are created.
+            :return:      An array of dicts with the down payment lines values.
+        """
+        self.ensure_one()
+
+        if self.advance_payment_method == 'percentage':
+            ratio = self.amount / 100
+        else:
+            ratio = self.fixed_amount / order.amount_total if order.amount_total else 1
```

Comment:
Should we add a warning if the fixed amount is bigger than the total amount ?

Reply:
There already is :)
![image](https://github.com/odoo/odoo/assets/3251665/9ef2a21f-73f0-43a5-9d60-8ab2435ac474)